go/vt/topotools/tablet.go

```diff
@@ -59,6 +60,45 @@ func ConfigureTabletHook(hk *hook.Hook, tabletAlias *topodatapb.TabletAlias) {
 	hk.ExtraEnv["TABLET_ALIAS"] = topoproto.TabletAliasString(tabletAlias)
 }
 
+// ChangeTags changes the tags of the tablet. Make this external, since these
+// transitions need to be forced from time to time.
+//
+// If successful, the updated tablet record is returned.
+func ChangeTags(ctx context.Context, ts *topo.Server, tabletAlias *topodatapb.TabletAlias, tabletTags map[string]string, replace bool) (*topodatapb.Tablet, error) {
+	var result *topodatapb.Tablet
+	_, err := ts.UpdateTabletFields(ctx, tabletAlias, func(tablet *topodatapb.Tablet) error {
+		if replace && maps.Equal(tablet.Tags, tabletTags) {
+			result = tablet
+			return topo.NewError(topo.NoUpdateNeeded, topoproto.TabletAliasString(tabletAlias))
+		}
+		if replace || tablet.Tags == nil {
+			tablet.Tags = tabletTags
+		} else {
+			var doUpdate bool
+			for key, val := range tabletTags {
+				if val == "" {
+					delete(tablet.Tags, key)
+					doUpdate = true
```

Comment:
nit: not super important, but it is possible the key doesn't exist in tablet.Tags either, so the updation might not be required.

Reply:
@GuptaManan100 good point! I believe I've resolved this now